src/benchmarks/blk.c

```diff
@@ -476,40 +439,85 @@ blk_exit(struct benchmark *bench, struct benchmark_args *args)
 	return 0;
 }
 
-static struct benchmark_info blk_read_info = {
-	.name		= "blk_read",
-	.brief		= "Benchmark for blk_read() operation",
-	.init		= blk_read_init,
-	.exit		= blk_exit,
-	.multithread	= true,
-	.multiops	= true,
-	.init_worker	= blk_init_worker,
-	.free_worker	= blk_free_worker,
-	.operation	= blk_operation,
-	.clos		= blk_clo,
-	.nclos		= ARRAY_SIZE(blk_clo),
-	.opts_size	= sizeof(struct blk_args),
-	.rm_file	= true,
-	.allow_poolset	= true,
-};
-
-REGISTER_BENCHMARK(blk_read_info);
-
-static struct benchmark_info blk_write_info = {
-	.name		= "blk_write",
-	.brief		= "Benchmark for blk_write() operation",
-	.init		= blk_write_init,
-	.exit		= blk_exit,
-	.multithread	= true,
-	.multiops	= true,
-	.init_worker	= blk_init_worker,
-	.free_worker	= blk_free_worker,
-	.operation	= blk_operation,
-	.clos		= blk_clo,
-	.nclos		= ARRAY_SIZE(blk_clo),
-	.opts_size	= sizeof(struct blk_args),
-	.rm_file	= true,
-	.allow_poolset	= true,
-};
+static struct benchmark_info blk_read_info;
+static struct benchmark_info blk_write_info;
 
-REGISTER_BENCHMARK(blk_write_info);
+CONSTRUCTOR(blk_costructor)
+void
+blk_costructor(void)
+{
+	blk_clo[0].opt_short = 'i';
```

Comment:
I suggest using a builder pattern (chaining setters) for this structure. Additionally, change the clos array into a vector, making it easier to use.

And IMHO, the main benchmark code should be aware of the benchmarks it supports. And instead of calling the constructor here (an overkill), we should iterate over an abstract benchmark classes and register them in a single place.